### ISSUE TITLE
scrape.php crashes

### ISSUE THREAD

**Issue by easyroot** [nZEDb/nZEDb#2451]
###Expected behaviour
The script should connect to IRC and populate the predb table (dev version)

###Actual behaviour
Upon the first insert into the database, scrape.php exits with this in php_errors.log

[07-May-2017 22:36:11 Europe/Berlin] PHP Fatal error:  Uncaught lithium\core\ConfigException: Missing key `id` from schema. in /var/www/nZEDb/libraries/lithium/data/Model.php:1050
Stack trace:
#0 /var/www/nZEDb/libraries/lithium/data/model/Query.php(649): lithium\data\Model::schema(NULL)
#1 /var/www/nZEDb/libraries/lithium/data/source/Database.php(1112): lithium\data\model\Query->schema()
#2 /var/www/nZEDb/libraries/lithium/data/model/Query.php(569): lithium\data\source\Database->fields(Array, Object(lithium\data\model\Query))
#3 /var/www/nZEDb/libraries/lithium/data/source/Database.php(645): lithium\data\model\Query->export(Object(lithium\data\source\database\adapter\MySql))
#4 /var/www/nZEDb/libraries/lithium/data/source/Database.php(595): lithium\data\source\Database->_queryExport(Object(lithium\data\model\Query))
#5 /var/www/nZEDb/libraries/lithium/aop/Filters.php(231): lithium\data\source\Database->lithium\data\source\{closure}(Array)
#6 /var/www/nZEDb/libraries/lithium/data/source/Database.php(632): lithium\aop\Filters::run(Object(lithium\data\source\databa in /var/www/nZEDb/libraries/lithium/data/Model.php on line 1050

Upon first encountering this issue I found that predb.id was actually named predb.ID but renaming the column did not change anything.

I've been going through the DB patches to see if I've missed something but haven't found anything obvious.

Reverting to an older commit removes the issue (cddfb4)

**Comment by Wally73** [comment]
make sure the predb table is like this
https://github.com/nZEDb/nZEDb/blob/dev/resources/db/schema/mysql-ddl.sql#L630

**Comment by easyroot** [comment]
I have tried dropping the table and recreating it from the DDL; even with an empty table it still fails.

**Comment by niel** [comment]
do: 
DESCRIBE predb;
and make sure the schema *exactly* matches the DDL. In particular be sure that the 'id' field is not 'ID' which is a problem we have seen from two others so far.

**Comment by niel** [comment]
We've only seen this with one (possibly two) others so far. In his case he had to export all his data and recreate the DB from scratch to fix the issue. Seems once the db has messed up with the wrong case it takes drastic measures to fix it. We're trying to find a less onerous cure for this, but there's no guarantees that we will.

**Comment by Faryn** [comment]
Ran into the same issue and simply renaming the column fixed it for me.

Statement for reference:
`ALTER TABLE predb CHANGE ID id INT(11) UNSIGNED NOT NULL AUTO_INCREMENT COMMENT 'Primary key';`

EDIT: It was still failing randomly with the same error message after that change, though less often.
I figured out I had the same issue (ID column instead of id) in the groups table. After fixing that too I haven't had any more errors.
